src/server/JasmineGraphInstanceService.cpp

```diff
@@ -1689,30 +1691,49 @@ map<long, double> calculateLocalPageRank(string graphID, double alpha, string pa
     // calculating local pagerank
     map<long, double> rankMap;
 
-    map<long, long> inDegreeDistribution;
+    std::map<long, long> inDegreeDistribution;
 
     std::string aggregatorFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
-    std::string iddFilePath = aggregatorFilePath + "/" + graphID + "_idd_" + partitionID;
-    ifstream dataFile;
-    dataFile.open(iddFilePath);
-
-    while (!dataFile.eof()) {
-        std::string str;
-        std::getline(dataFile, str);
-        std::stringstream buffer(str);
-        std::string temp;
-        std::vector<long> values;
-
-        while (getline(buffer, temp, '\t')) {
-            values.push_back(::strtod(temp.c_str(), nullptr));
-        }
-        if (values.size() == 2) {
-            long nodeID = values[0];
-            long iddValue = values[1];
-            inDegreeDistribution.insert(std::make_pair(nodeID, iddValue));
+
+    std::string partitionCount = Utils::getJasmineGraphProperty("org.jasminegraph.server.npartitions");
+    int parCount = std::stoi(partitionCount);
+
+    for (int partitionID = 0; partitionID < parCount; ++partitionID) {
+        std::string iddFilePath = aggregatorFilePath + "/" + graphID + "_idd_" + std::to_string(partitionID);
+        std::ifstream dataFile;
+        dataFile.open(iddFilePath);
+
+        while (!dataFile.eof()) {
+            std::string str;
```

Comment:
```suggestion
            std::string line;
```

Reply:
Refactor the remaining occurrences of `str` with `line`.

Reply:
Done.

Reply:
Ok